File: personalizatio-sdk/src/personaclick/java/com/personaclick/sdk/Personaclick.java
package com.personaclick.sdk;

import android.content.Context;

import com.personalizatio.SDK;
import com.personalizatio.BuildConfig;

/**
 * Created by Sergey Odintsov
 *
 * @author nixx.dj@gmail.com
 */
final public class Personaclick extends SDK {

	public static final String TAG = "PERSONACLICK";
	public static final String NOTIFICATION_TYPE = "PERSONACLICK_NOTIFICATION_TYPE";
	public static final String NOTIFICATION_ID = "PERSONACLICK_NOTIFICATION_ID";
	protected static final String PREFERENCES_KEY = "personaclick.sdk";
	protected static final String API_URL = BuildConfig.DEBUG ? "https://api.rees46.ru/" : "https://api.personaclick.com/";
	/**
	 * @param context application context
	 * @param shop_id Shop key
	 */
	private Personaclick(Context context, String shop_id) {
		super(context, shop_id, API_URL, TAG, PREFERENCES_KEY, "android");
	}

	/**
	 * Initialize api
	 * @param shop_id Shop key
	 */
	public static void initialize(Context context, String shop_id) {
		if( instance == null ) {
			instance = new Personaclick(context, shop_id);
		}
	}
}

File: personalizatio-sdk/src/personaclick/java/com/personaclick/sdk/Personaclick.kt
package com.personaclick.sdk

import android.content.Context
import com.personalizatio.BuildConfig
import com.personalizatio.SDK

class Personaclick private constructor() : SDK() {

    companion object {
        const val TAG: String = "PERSONACLICK"
        const val NOTIFICATION_TYPE: String = "PERSONACLICK_NOTIFICATION_TYPE"
        const val NOTIFICATION_ID: String = "PERSONACLICK_NOTIFICATION_ID"
        protected const val PREFERENCES_KEY: String = "personaclick.sdk"
        protected val API_URL: String = if (BuildConfig.DEBUG) "http://192.168.1.8:8080/" else "https://api.personaclick.com/"

        /**
         * Initialize api
         * @param context application context
         * @param shopId Shop key
         */
        fun initialize(context: Context, shopId: String) {
            if (!isInstanced()) {
                getInstance().initialize(context, shopId, API_URL, TAG, PREFERENCES_KEY, "android")
            }
        }
    }
}
